src/ore/src/retry.rs

```diff
@@ -24,48 +24,45 @@
 //! Retry a contrived fallible operation until it succeeds:
 //!
 //! ```
-//! # tokio::runtime::Runtime::new().unwrap().block_on(async {
 //! use std::time::Duration;
 //! use ore::retry::Retry;
 //!
-//! let res = Retry::default().retry(|state| async move {
+//! let res = Retry::default().retry(|state| {
 //!    if state.i == 3 {
 //!        Ok(())
 //!    } else {
 //!        Err("contrived failure")
 //!    }
-//! }).await;
+//! });
 //! assert_eq!(res, Ok(()));
-//! # });
 //! ```
 //!
 //! Limit the number of retries such that success is never observed:
 //!
 //! ```
-//! # tokio::runtime::Runtime::new().unwrap().block_on(async {
 //! use std::time::Duration;
 //! use ore::retry::Retry;
 //!
-//! let res = Retry::default().max_tries(2).retry(|state| async move {
+//! let res = Retry::default().max_tries(2).retry(|state| {
 //!    if state.i == 3 {
 //!        Ok(())
 //!    } else {
 //!        Err("contrived failure")
 //!    }
-//! }).await;
+//! });
 //! assert_eq!(res, Err("contrived failure"));
-//! # });
+//! ```
 
 use std::cmp;
 use std::future::Future;
 use std::pin::Pin;
 use std::task::{Context, Poll};
-use std::time::Instant;
+use std::thread;
 
 use futures::{ready, Stream, StreamExt};
 use pin_project::pin_project;
 use tokio::io::{AsyncRead, ReadBuf};
-use tokio::time::Duration;
+use tokio::time::{self, Duration, Instant, Sleep};
 
 // TODO(benesch): remove this if the `duration_constants` feature stabilizes.
```

Comment:
Check out https://github.com/rust-lang/rust/pull/84120

Reply:
zomg!

Reply:
I actually went to see if `duration_constants` was stabilized, saw that it hadn't been, and got sad. Did not think to see if `Duration::MAX` had escaped separately! Tyvm!